script.js

```diff
@@ -57,29 +57,14 @@ class Snake {
     });
   }
 
-  move = () => {
-    game.checkKeys();
-
+  getNextPosition = () => {
     const delta = Snake.DELTAS[this.direction];
     const x = this.elements[0].x + delta.x;
     const y = this.elements[0].y + delta.y;
-    
-    if (game.snake.elements[0].x == game.apples[0].x && game.snake.elements[0].y == game.apples[0].y) {
-      game.removeApple();
-    }
-    
-    if (x >= game.field.width || y == game.field.height || x < 0 || y < 0) {
-      game.gameOver();
-      return
-    }
-
-    for (let i = 1; i < game.snake.elements.length; i++) {
-      if (x  === game.snake.elements[i].x && y  === game.snake.elements[i].y) {
-        game.gameOver();
-        return
-      }
-    }
+    return [x, y];
```

Comment:
Would probably be better to return an object with `x` and `y` as properties:
```suggestion
    return {x, y};
```
You can then access those values by `position.x` and `position.y` instead of `[0]` and `[1]` (lines 166-167 below) or even pass the object to `isValidMove` and `move`.